plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.sdk;
+
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+import com.google.cloud.tools.eclipse.sdk.internal.CloudSdkInstallJob;
+import com.google.cloud.tools.managedcloudsdk.ManagedCloudSdk;
+import com.google.cloud.tools.managedcloudsdk.ManagedSdkVerificationException;
+import com.google.cloud.tools.managedcloudsdk.ManagedSdkVersionMismatchException;
+import com.google.cloud.tools.managedcloudsdk.UnsupportedOsException;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponent;
+import java.nio.file.Path;
+import org.eclipse.core.runtime.jobs.Job;
+
+public class CloudSdkManager {
+
+  /**
+   * Returns {@code CloudSdk} if there exists an up-to-date Cloud SDK with the App Engine Java
+   * component installed at the managed location. Otherwise, throws {@code
+   * CloudSdkNotReadyException} after triggering a background installation task. Callers who get the
+   * exception may need to try this method at later points.
+   *
+   * Note that even if the Cloud SDK is already installed, this method executes {@code gcloud} twice
+   * to check components and versions, which may incur a noticeable delay. If possible, avoid
+   * unnecessary repetitive calls.
```

Comment:
Shouldn't this multiple-invocation be an internal detail?  On the first fetch, we may perform some checks, but no checks should be required on subsequent fetches.